exec/vector/src/main/java/org/apache/drill/exec/util/DecimalUtility.java

```diff
@@ -320,25 +203,25 @@ public static void getSparseFromBigDecimal(BigDecimal input, ByteBuf data, int s
 
     int destIndex = nDecimalDigits - roundUp(scale) - 1;
 
-    // we use base 1 billion integer digits for out integernal representation
+    // we use base 1 billion integer digits for out internal representation
     BigDecimal base = new BigDecimal(DIGITS_BASE);
 
-    while (integerPart.compareTo(BigDecimal.ZERO) == 1) {
-        // store the modulo as the integer value
-        data.setInt(startIndex + (destIndex * INTEGER_SIZE), (integerPart.remainder(base)).intValue());
-        destIndex--;
-        // Divide by base 1 billion
-        integerPart = (integerPart.divide(base)).setScale(0, BigDecimal.ROUND_DOWN);
+    while (integerPart.compareTo(BigDecimal.ZERO) > 0) {
```

Comment:
Not saying, this change is not correct but `compareTo` for BidDecimal returns 1.